src/Core/Silk.NET.BuildTools/Converters/Readers/OpenCLReader.cs

```diff
@@ -835,42 +755,167 @@ public IEnumerable<Enum> ReadEnums(object obj, BindTask task)
 
             var registry = doc.Element("registry");
             Debug.Assert(registry != null, $"{nameof(registry)} != null");
-            
-            var allEnums = registry
+
+            var apis = registry.Elements("feature").Concat
+            (
+                registry.Elements("extensions").Elements("extension")
+                ?? throw new InvalidDataException()
+            );
+            Debug.Assert(apis != null, nameof(apis) + " != null");
+
+            var ulongEnums = new HashSet<string>(registry
+                .Elements("types")
+                .Elements("type")
+                .Where(e => e.Elements("type").SingleOrDefault()?.Value == "cl_bitfield")
+                .Select(e => Rename(e.Element("name").Value, task).Renamed));
+
+            var flagEnums = new HashSet<string>(registry
+                .Elements("enums")
+                .Where(e => e.Attribute("type")?.Value == "bitmask")
```

Comment:
Apologies for coming back with comments i didn't pick up on before... don't bitfields also constitute the flags attribute?